### ISSUE TITLE
[Feature Request] Search

### ISSUE THREAD

**Issue by claxonberlin** [wawanbreton/CuraOnshapePlugin#2]
First off, fantastic job. It speeds up workflow a lot and deprecates unnecessary STL downloads. So far, everything works perfectly using Cura 5.8.1. I couldn't find how it worked immediately, though. Found it under Open File eventually, so might be a good idea to add a screenshot or description of that as well, since people love screenshots :)

Issue: Since I have quite a lot of folders and documents in my account, it's hard to find what I need quickly. Do you think a search function could be added?

**Comment by claxonberlin** [comment]
@wawanbreton not sure if you saw this.

**Comment by wawanbreton** [comment]
> @wawanbreton not sure if you saw this.

No indeed, thanks for pinging me ! I have changed my notification settings wo I should get them for the next issues.

> First off, fantastic job. It speeds up workflow a lot and deprecates unnecessary STL downloads

Thanks :slightly_smiling_face: I do use it myself very often and it really makes my life easier, especially when doing live modifications.

> I couldn't find how it worked immediately, though. Found it under Open File eventually, so might be a good idea to add a screenshot or description of that as well, since people love screenshots :)

That is very doable. Do you think the README would be the right place ? I assume this is the first thing people see when landing on GitHub.

> Issue: Since I have quite a lot of folders and documents in my account, it's hard to find what I need quickly. Do you think a search function could be added?

That would be interesting indeed. I usually keep my documents organized in folders in a tree, but that is still a lot of them, so I can totally understand that finding a specific document can be hard. I can propose 2 solutions:
1. Search through the currently displayed list. If you are at the home level, it will only filter amongst the displayed folders, but will not look for the content inside them. That is a solution that I know would be very easy to implement, but maybe not very helpful.
2. Search through either folders names, documents, part studios, and parts. That would allow jumping directly to a part if you know its name, so pretty convenient. However, I don't know if the Onshape API provides such a function, so it may not be possible. If they provide it, it is also more work on my side because I currently only display elements of the same type at some point. But that would be doable though. After a quick search, the API provides a search function, but I don't know exactly what it returns. I'll have to do some tests.

Can you tell me if #1 would be of any use in your case, or if it would really be pointless ?

**Comment by wawanbreton** [comment]
I confirm that the search API allows for searching anything, probably like the search button that is on the Onshape main page. So it must be doable :slightly_smiling_face:

**Comment by claxonberlin** [comment]
> Do you think the README would be the right place ?

Definitely. A little gif of the Cura interface where your mouse moves through the menus and ends when the Onshape popup shows would be perfect. Or even just this screenshot, haha: 
![image](https://github.com/user-attachments/assets/d75958b0-9215-44ea-a96e-3221662bc506)

> Can you tell me if https://github.com/wawanbreton/CuraOnshapePlugin/issues/1 would be of any use in your case, or if it would really be pointless ?

It could prove useful, but the user would have to search multiple times, right? It would look like:
1. Search name of Folder your document is in (assuming "Open From OnShape" always starts on the home page<sup>1</sup>)
2. Click the Folder to enter it
3. Search the Folder for the right Document
4. Click the Document to enter it
5. Search the Document for the right part name
6. Click the part to add it to Cura

That's quite a lot of steps in the case of large assemblies and industry level organisation. But it would definitely be a big step up of having no search, so if it's very easy to implement I'd go for it. But #2 would definitely be preferable depending on performance.

<sup>1</sup>A setting could be to open the window where you left it - handy for repeat prints where you edit your part and move between OS and Cura a lot.
